crates/aptos-crypto/src/p256/p256_keys.rs

```diff
@@ -0,0 +1,270 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! This file implements traits for P256 private keys and public keys.
+
+#[cfg(any(test, feature = "fuzzing"))]
+use crate::test_utils::{self, KeyPair};
+use crate::{
+    p256::{P256Signature, P256_PRIVATE_KEY_LENGTH, P256_PUBLIC_KEY_LENGTH},
+    hash::CryptoHash,
+    traits::*,
+};
+use p256::ecdsa::signature::Signer;
+use p256;
+use aptos_crypto_derive::{DeserializeKey, SerializeKey, SilentDebug, SilentDisplay};
+use core::convert::TryFrom;
+#[cfg(any(test, feature = "fuzzing"))]
+use proptest::prelude::*;
+use serde::Serialize;
+use std::fmt;
+
+/// A P256 private key
+#[derive(DeserializeKey, SerializeKey, SilentDebug, SilentDisplay)]
+pub struct P256PrivateKey(pub(crate) p256::ecdsa::SigningKey);
+
+impl private::Sealed for P256PrivateKey {}
+
+#[cfg(feature = "assert-private-keys-not-cloneable")]
+static_assertions::assert_not_impl_any!(P256PrivateKey: Clone);
+
+#[cfg(any(test, feature = "cloneable-private-keys"))]
+impl Clone for P256PrivateKey {
+    fn clone(&self) -> Self {
+        let serialized: &[u8] = &(self.to_bytes());
+        P256PrivateKey::try_from(serialized).unwrap()
+    }
+}
+
+/// A P256 public key
+#[derive(DeserializeKey, Clone, SerializeKey)]
+pub struct P256PublicKey(pub(crate) p256::ecdsa::VerifyingKey);
+
+impl private::Sealed for P256PublicKey {}
```

Comment:
should this be in `traits.rs` where other implementations are also specified?

Reply:
`P256PublicKey`? No, `traits.rs` only contains traits that are agnostic to the signature scheme used, while these structs are specific to ECDSA over P256

Reply:
Oh you meant `Sealed`. I think that makes sense to put it in `traits.rs` with the rest

Reply:
Moved Sealed to `traits.rs`